dkist/dataset/utils.py

```diff
@@ -19,18 +21,23 @@ def dataset_info_str(ds):
         ds = ds[0, 0]
     wcs = ds.wcs.low_level_wcs
 
-    # Pixel dimensions table
+    # Array dimensions table
 
-    instr = ds.inventory.get("instrument", "")
+    instr = ds.inventory.get("instrumentName", "")
     if instr:
         instr += " "
+    nframes = ds.inventory.get("frameCount", "an unknown number")
+    dsID = ds.inventory.get("datasetId", "(no DatasetID)")
 
     if is_tiled:
-        s = f"This {dstype} consists of an array of {tile_shape} Dataset objects\n\n"
-        s += f"Each {instr}Dataset has {wcs.pixel_n_dim} pixel and {wcs.world_n_dim} world dimensions\n\n"
+        s = f"This {instr}{dstype} {dsID} consists of an array of {tile_shape} Dataset objects\n\nEach Dataset "
     else:
-        s = f"This {instr}Dataset has {wcs.pixel_n_dim} pixel and {wcs.world_n_dim} world dimensions\n\n"
-    s += f"{ds.data}\n\n"
+        s = f"This {instr}Dataset {dsID} "
+
+    s += f"has {wcs.pixel_n_dim} pixel and {wcs.world_n_dim} world dimensions and consists of {nframes} frames\n"
```

Comment:
I would punt nframes into the `if ds.files`  and use `len(ds.files)` instead of getting it out of inventory as then it will update properly when sliced.

Reply:
Each individual tile would be three (presumably you have indexed down into a single dataset there), the total number of files for a tiled dataset would be `np.sum([len(ds.files) for ds in tds.flat])`.

Reply:
Right. But my point is, which of these equally legitimate things do we want to present to the user as the "number of frames"?

Reply:
I would say the latter one, as it again reflects what happens if you use the new slicing machinery on `TiledDataset`.